cumulusci/cli/cci.py

```diff
@@ -259,42 +253,93 @@ def version():
     click.echo()
 
 
-@main.command(name="shell", help="Drop into a Python shell")
-def shell():
-    try:
-        config = load_config(load_project_config=True, load_keychain=True)
-    except SystemExit:
-        config = load_config(load_project_config=False, load_keychain=False)
+@cli.command(name="shell", help="Drop into a Python shell")
+@pass_runtime(require_project=False)
+def shell(runtime):
+    # alias for backwards-compatibility
+    config = runtime
 
     code.interact(local=dict(globals(), **locals()))
 
 
+GIST_404_ERR_MSG = """A 404 error code was returned when trying to create your gist.
+Please ensure that your GitHub personal access token has the 'Create gists' scope."""
+CCI_LOG_NOT_FOUND_MSG = """No logfile to open at path: {}
+Please ensure you're running this command from the same directory you were experiencing an issue."""
+LAST_CMD_HEADER = "\n\n\nLast Command Run\n================================\n"
+CCI_LOGFILE_PATH = Path.home() / ".cumulusci/logs/cci.log"
+
+
+@cli.command(name="gist", help="Create a GitHub gist from the latest logfile")
+@pass_runtime(require_project=False)
+def gist(runtime):
+    try:
+        last_cmd_log = open(CCI_LOGFILE_PATH, "r")
```

Comment:
Use open as a context manager to make sure the file gets closed when we're done with it.